File: Python/lt_01_array/lt_01_array_07.py
import random


def makelista():
    lista = []
    for i in range(20):
        lista.append(random.randint(0, 1000))

    print(lista)

    for i in range(len(lista)):
        for j in range(i + 1, len(lista)):
            if lista[i] > lista[j]:
                aux = lista[i]
                lista[i] = lista[j]
                lista[j] = aux

    print(lista)

    return lista


def findvet(lista, min, max, find):
    if min < max:
        meio = int(((max - min) / 2) + min)
        print(min, meio, max, "|")
        if lista[meio] > find:
            findvet(lista, min, meio, find)
        elif lista[meio] < find:
            findvet(lista, meio, max, find)
        else:
            print("Valor Encontrado na casa", meio + 1, ".\n")
    else:
        print("Valor não Encontrado")


lista = makelista()
findvet(lista, 0, len(lista), int(input("Insira o valor a ser procurado:")))

File: C/lt_01_array/lt_01_array_07.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void search(int *vet, int min, int max, int find) {
  int media = (((max - min) / 2) + min);

  if (min != media) {
    int media = (((max - min) / 2) + min);
    if (*(vet + media) == find) {

      printf("O valor %d foi encontrado!\n", find);
    } else if (*(vet + media) < find) {

      printf("|%d", *(vet + media));
      printf("|%d|%d|%d|\n", min, media, max);
      search(vet, media, max, find);

    } else if (*(vet + media) > find) {

      printf("|%d", *(vet + media));
      printf("|%d|%d|%d|\n", min, media, max);
      search(vet, min, media, find);
    }
  } else {
    if (*(vet + media) == find) {

      printf("O valor %d foi encontrado!\n", find);
    } else {

      printf("O valor %d não foi encontrado!\n", find);
    }
  }
}

int main() {
  srand(time(NULL));
  int i, j, aux, vet[6];

  for (i = 0; i < (sizeof(vet) / sizeof(vet[0])); i++) {
    vet[i] = rand() % 100;
    printf("|%d", vet[i]);
  }
  printf("|\n");

  for (i = 0; i < (sizeof(vet) / sizeof(vet[0])); i++) {
    for (j = i; j < (sizeof(vet) / sizeof(vet[0])); j++) {
      if (vet[i] > vet[j]) {
        aux = vet[i];
        vet[i] = vet[j];
        vet[j] = aux;
      }
    }
  }

  for (i = 0; i < (sizeof(vet) / sizeof(vet[0])); i++) {
    printf("|%d", vet[i]);
  }
  printf("|\n");

  printf("Insira o valor a ser procurado: ");
  scanf("%d", &aux);
  search(vet, 0, (sizeof(vet) / sizeof(vet[0])), aux);

  return 0;
}
